src/folding/nova/circuits.rs

```diff
@@ -226,8 +236,8 @@ pub trait FCircuit<F: PrimeField>: Copy {
 pub struct AugmentedFCircuit<C: CurveGroup, FC: FCircuit<CF1<C>>> {
     pub poseidon_config: PoseidonConfig<CF1<C>>,
     pub i: Option<CF1<C>>,
-    pub z_0: Option<C::ScalarField>,
-    pub z_i: Option<C::ScalarField>,
+    pub z_0: Option<Vec<C::ScalarField>>,
+    pub z_i: Option<Vec<C::ScalarField>>,
```

Comment:
I'm starting to thing that we should have some kind of builder pattern for this. As it has soo many fields already and most of them are `None` a lot of times.

Reply:
Ohh the amend wasn't even needed no worries. You're right that only testing gets affected. So fair point.

Thanks anyways!